src/python/zquantum/qaoa/maxcut.py

```diff
@@ -2,37 +2,103 @@
 import numpy as np
 from itertools import combinations
 from random import uniform
-
-
 from openfermion import QubitOperator
 from zquantum.core.utils import dec2bin
-from zquantum.core.graph import generate_graph_node_dict
+from zquantum.core.graph import generate_graph_node_dict, generate_graph_from_specs
+from typing import Dict, List, Union
+
+from .farhi_ansatz import QAOAFarhiAnsatz
+
+
+def create_farhi_qaoa_circuits(
+    hamiltonians: List[QubitOperator], number_of_layers: int
+):
+    """Creates parameterizable quantum circuits based on the farhi qaoa ansatz for each 
+    hamiltonian in the input list using the set number of layers. 
+
+    Args:
+        hamiltonians (List[QubitOperator]): List of hamiltonians for constructing the
+            circuits
+        number_of_layers (int): The number of layers of the ansatz in the circuit
+                
+    Returns:
+        List of zquantum.core.circuit.Circuit
+    """
+    circuitset = []
+    for hamiltonian in hamiltonians:
+        ansatz = QAOAFarhiAnsatz(number_of_layers, hamiltonian)
+        circuitset.append(ansatz.parametrized_circuit)
+    return circuitset
+
+
+def get_random_maxcut_hamiltonians(
+    graph_specs: Dict,
+    number_of_instances: int,
+    number_of_qubits: Union[int, List[int]],
```

Comment:
Actually, I would rename it to "possible_numbers_of_qubits" and allow only for list.
This ambiguity is confusing :)

Reply:
Done! Commit 439a707902cc4fcae3fcffdaa112793e59dfe1f6 and update of the unittest in 4211dfb30c89a6de2ed2341f674b46a0b117b74e